vertexai/internal/sentencepiece/token.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sentencepiece
+
+import "fmt"
+
+// Token represents a single token from the input text. ID is a unique token
+// identifier that the model uses in its internal representation. Text is
+// the piece of text this token represents.
+type Token struct {
```

Comment:
There's some invariant here, like in the set of Tokens from a given call to Encode, all tokens with the same ID will have the same text, right?

OOC (no need to do this), if that's true, could we save space by having Encode return a ([]int, map[int]string)?

Reply:
Looking more closely, it seems that an Encoder instance defines all possible token IDs, so every call to Encode uses the same mapping.

I think that's worth documenting somewhere. Maybe on Encoder.

Reply:
Yes, I'll add a documentation. This is read from the proto.

The mapping is a bit trickier becasue of reserved symbols (see `symbolToID`). I expect both token ids and values to be useful, so `Encode` does both (space isn't much at premium here, but speed is) - otherwise the mapping would be complex for users - we'd have to expose something similar to `symbolToID`

Reply:
Ack.